extensions/android-paging3-rx3/src/main/java/com/squareup/sqldelight/android/paging3/RxOffsetQueryPagingSource.kt

```diff
@@ -0,0 +1,59 @@
+package com.squareup.sqldelight.android.paging3
+
+import androidx.paging.PagingState
+import androidx.paging.rxjava3.RxPagingSource
+import com.squareup.sqldelight.Query
+import com.squareup.sqldelight.Transacter
+import io.reactivex.rxjava3.core.Scheduler
+import io.reactivex.rxjava3.core.Single
+import kotlinx.coroutines.rx3.asCoroutineDispatcher
+import kotlinx.coroutines.rx3.rxSingle
+
+internal class RxOffsetQueryPagingSource<RowType : Any>(
+  queryProvider: (limit: Long, offset: Long) -> Query<RowType>,
+  countQuery: Query<Long>,
+  transacter: Transacter,
+  scheduler: Scheduler,
+) : RxPagingSource<Long, RowType>() {
+
+  private val delegate = QueryPagingSource(
+    countQuery = countQuery,
+    transacter = transacter,
+    queryProvider = queryProvider,
+    dispatcher = scheduler.asCoroutineDispatcher(),
+  )
+
+  override fun getRefreshKey(state: PagingState<Long, RowType>) = delegate.getRefreshKey(state)
+
+  override fun loadSingle(params: LoadParams<Long>): Single<LoadResult<Long, RowType>> = rxSingle {
+    delegate.load(params)
+  }
+}
+
+/**
+ * Create a [RxPagingSource] that pages through results according to queries generated by
+ * [queryProvider]. Queries returned by [queryProvider] should expect to do SQL offset/limit
+ * based paging. For that reason, [countQuery] is required to calculate pages and page offsets.
+ *
+ * An example query returned by [queryProvider] could look like:
+ *
+ * ```sql
+ * SELECT value FROM numbers
+ * LIMIT 10
+ * OFFSET 100;
+ * ```
+ *
+ * Queries will be executed on [scheduler].
+ */
+@Suppress("FunctionName")
+fun <RowType : Any> RxQueryPagingSource(
+  countQuery: Query<Long>,
+  transacter: Transacter,
+  scheduler: Scheduler,
```

Comment:
This is the only Rx thing in the public API inside this module. What's the value in this module instead of forcing the caller to call `scheduler.asCoroutineDispatcher()` and having them use the core paging3 module?

Reply:
Honestly, after viewing how streams of pages are exposed [here](https://cs.android.com/androidx/platform/tools/dokka-devsite-plugin/+/master:testData/paging/source/androidx/paging/rxjava3/PagingRx.kt;l=17?q=PagingRx&sq=), I'm not so sure! I'm mainly just going off of Alec's guidance to support both rx3 and coroutines. 

@AlecStrong thoughts?

Reply:
it exposes `RxPagingSource` in the public API as well which might be the value. Unless downstream theres a straightforward way to go from the coroutines data source to an rx one in which case you'd just do the conversion there. I don't feel that strongly, I assume the implementation of paging3 means its going to be coroutines under the hood anyway so its just a question of where to do the conversion if you're doing rx and how clunky it is

Reply:
all the `RxPagingSource` does is calls `Single.await` from `suspend fun load(...)`.

One thing which might be weird for non-coroutines users. If my codebase heavily uses rx3 and I want to use Paging, I'd see that an rx3 artifact is offered. However, SQLDelight doesn't offer an out of the box implementation. That would require our end users to explicitly depend on the coroutines version of paging as well as the coroutines-rx3 compat artifact to bridge a `Scheduler` to a `CoroutineDispatcher`. 

When using the rx paging library, these dependencies still exist, however they're hidden from the user with `implementation`.

Reply:
yea - i think that _might_ be worth it? i dont feel super strongly, curious jake if you do

Reply:
RxPaging source is a thing if _you_ are implementing paging on top of something and you are using Rx. But in both the Rx case and the coroutine case you are dealing with SQLDelight `Query`. So the only thing that's different is whether you pass a `Dispatcher` or a `Scheduler` and I don't think that's enough value.

There's no value in returning a `RxPagingSource` (and in fact we should probably be returning a vanilla `PagingSource`) because all you do is hand it off to the paging library which interacts with it as a coroutine-based `PagingSource`.

I think we set a default dispatcher of `Dispatchers.IO` in the core paging extension and then you never see either a coroutine API or an Rx API unless you need to customize the dispatcher (which will be a rare case).

Reply:
I'm glad you pointed this out -- I agree. I think an Rx API should be as simple as:

```kotlin
@Suppress("FunctionName")
fun <RowType : Any> QueryPagingSource(
  countQuery: Query<Long>,
  transacter: Transacter,
  scheduler: Scheduler,
  queryProvider: (limit: Long, offset: Long) -> Query<RowType>,
): PagingSource<Long, RowType> = OffsetQueryPagingSource(
  queryProvider,
  countQuery,
  transacter,
  scheduler.asCoroutineDispatcher(),
)
```

Reply:
Yeah. And I don't think that meets the bar of requiring an entire module. Because you could just as easily include a dependency on coroutines-rxN and do the `asCoroutineDispatcher()` call for basically the same amount of code.

Reply:
Cool, I'm going to go ahead and close this PR.